crates/stark-felt/src/lib.rs

```diff
@@ -161,11 +171,84 @@ impl Felt {
         Self(self.0.pow(exponent))
     }
 
-    /// Performs self modulo n, with n being smaller that the stark field prime.
-    pub fn mod_floor(&self, n: &Self) -> Self {
-        Self(FieldElement::from(
-            &(self.0).representative().div_rem(&n.0.representative()).1,
-        ))
+    pub fn pow_felt(&self, exponent: &Felt) -> Self {
+        Self(self.0.pow(exponent.0.representative()))
+    }
+
+    /// Modular multiplication.
+    pub fn mul_mod(&self, rhs: &Self, p: &NonZeroFelt) -> Self {
+        (self * rhs).div_rem(p).1
+    }
+
+    /// Modular multiplicative inverse.
+    pub fn inverse_mod(&self, p: &NonZeroFelt) -> Option<Self> {
+        self.inverse().map(|x| x.div_rem(p).1)
+    }
+
+    pub fn mod_floor(&self, n: &NonZeroFelt) -> Self {
+        self.div_rem(n).1
+    }
+
+    pub fn to_u32(&self) -> Option<u32> {
+        self.to_u64().and_then(|n| n.try_into().ok())
+    }
+
+    pub fn to_u64(&self) -> Option<u64> {
```

Comment:
Shouldn't we use this trait for those functions ? https://docs.rs/num/latest/num/trait.ToPrimitive.html